### ISSUE TITLE
[Bug]: Some accessibility issues

### ISSUE THREAD

**Issue by zhou-zhoukang** [Pool-Of-Tears/GreenStash#170]
### Steps to reproduce

Hi there! I really love the GreenStash app — it's clear that a lot of thought and care has gone into its UI. Thank you for your hard work on this great open source project. However, I noticed a few areas where accessibility could be improved.

1. The red boxes in the image below indicate all the UI elements that can be accessed by visually impaired users using screen readers. The components are quite fragmented, requiring blind users to swipe right multiple times to obtain information sequentially. If each setting item could be grouped into a single accessibility unit, it would be greatly helpful. 

![Image](https://github.com/user-attachments/assets/b887d18e-97df-4f83-ae1f-8765cf6d27d3)

2.  The "Back" button is missing ContentDescriptions. Without content descriptions, users with visual impairments may have difficulty understanding the purpose of these buttons when using screen readers.

![Image](https://github.com/user-attachments/assets/984f1455-aedc-4ff0-9e0c-ceb4fe00158f)

3. Adding an `onClickLabel` to the Button might also be a good practice. An example of how this can be implemented is provided in the code snippet below. For instance, when focusing a Button, a screen reader might announce "Double tap to **activate**" before a label is added. After adding an appropriate label, it would instead say "Double tap to **Edit**", which can make the button's function more intuitive and clear to users. 
```
modifiler = Modifier.semantics {  
   onClick(label = "Edit") { }
}
```

### Expected behaviour

A potential approach would be using Compose's semantic APIs to group the components together, as demonstrated in Compose's official accessibility [documentation](https://developer.android.com/develop/ui/compose/accessibility/merging-clearing).   Specifically, adding `Modifier.semantics(mergeDescendants = true)` to the Row component in the `SettingsItem` function may be helpful.

### Actual behaviour

Already mentioned in the “Steps to reproduce” section.

### Android version

Android 12

### GreenStash version

3.9.2

**Comment by starry-shivam** [comment]
Interesting suggestions. As someone who is somewhat visually impaired myself, I try to add as many accessibility features as possible. However, I admit that these are usually a secondary priority for me. I’ll try to implement your suggestions to improve the accessibility features the next time I get to work on this app, but If you’d like to contribute these changes yourself, you’re very welcome to do so. Thanks!

**Comment by zhou-zhoukang** [comment]
Thank you for taking the time to address the issue. I'm a student focusing on accessibility issues in Android apps, especially with new frameworks like Jetpack Compose and their impact on accessibility features. Your app has truly set a great example in terms of software accessibility. I would be really honored to contribute to your project, and I will try my very best.

**Comment by starry-shivam** [comment]
Closing this issue since the PR to fix it has been merged successfully.
